oauthproxy.go

```diff
@@ -108,17 +109,14 @@ type OAuthProxy struct {
 	PassAccessToken         bool
 	SetAuthorization        bool
 	PassAuthorization       bool
+	SkipAuthStripHeaders    bool
 	PreferEmailToUser       bool
-	skipAuthRegex           []string
-	skipAuthPreflight       bool
-	skipAuthStripHeaders    bool
 	skipJwtBearerTokens     bool
 	mainJwtBearerVerifier   *oidc.IDTokenVerifier
 	extraJwtBearerVerifiers []*oidc.IDTokenVerifier
-	compiledRegex           []*regexp.Regexp
 	templates               *template.Template
 	realClientIPParser      ipapi.RealClientIPParser
-	trustedIPs              *ip.NetSet
+	allowlists              []allowlist.Allowlist
```

Comment:
We talked before about how `allowlist` doesn't best capture what we're using it for.  The words `trusted` and `bypass` come to mind more.  Then again, it's a pretty standard term, so it might be better anyways.  Definetly something to think about picking the most relevant name for, especially considering the scope of this mechaanism.